jwql/instrument_monitors/common_monitors/bias_monitor.py

```diff
@@ -0,0 +1,554 @@
+    these files. The monitor will extract the 0th group from the new
+    dark files and output the contents into a new file located in
+    a working directory. It will then perform statistical measurements
+    on these files before and after pipeline calibration in order to
+    monitor the bias levels over time as well as ensure the pipeline
+    superbias is sufficiently calibrating new data. Results are all
+    saved to database tables.
+
+    Attributes
+    ----------
+    output_dir : str
+        Path into which outputs will be placed
+
+    data_dir : str
+        Path into which new dark files will be copied to be worked on
+
+    query_start : float
+        MJD start date to use for querying MAST
+
+    query_end : float
+        MJD end date to use for querying MAST
+
+    instrument : str
+        Name of instrument used to collect the dark current data
+
+    aperture : str
+        Name of the aperture used for the dark current (e.g.
+        ``NRCA1_FULL``)
+    """
+
+    def __init__(self):
+        """Initialize an instance of the ``Bias`` class.
+        """
+
+    def collapse_image(self, image):
+        """Median-collapse the rows and columns of an image.
+
+        Parameters
+        ----------
+        image : numpy.ndarray
+            2D array on which to calculate statistics
+
+        Returns
+        -------
+        collapsed_rows : numpy.ndarray
+            1D array of the collapsed row values
+        
+        collapsed_columns : numpy.ndarray
+            1D array of the collapsed column values
+        """
+
+        collapsed_rows = np.nanmedian(image, axis=1)
+        collapsed_columns = np.nanmedian(image, axis=0)
+
+        return collapsed_rows, collapsed_columns
+
+    def extract_zeroth_group(self, filename):
+        """Extracts the 0th group of a fits image and outputs it into
+        a new fits file.
+        
+        Parameters
+        ----------
+        filename : str
+            The fits file from which the 0th group will be extracted.
+
+        Returns
+        -------
+        output_filename : str
+            The full path to the output file
+        """
+
+        output_filename = os.path.join(self.data_dir,
+            os.path.basename(filename).replace('.fits', '_0thgroup.fits'))
+        
+        # Write a new fits file containing the primary and science
+        # headers from the input file, as well as the 0th group
+        # data of the first integration
+        if not os.path.isfile(output_filename):
+            hdu = fits.open(filename)
+            new_hdu = fits.HDUList([hdu['PRIMARY'], hdu['SCI']])
+            new_hdu['SCI'].data = hdu['SCI'].data[0:1, 0:1, :, :]
+            new_hdu.writeto(output_filename)
+            hdu.close()
+            new_hdu.close()
+            logging.info('\t{} created'.format(output_filename))
+        else:
+            logging.info('\t{} already exists'.format(output_filename))
+            pass
```

Comment:
I don't think this pass statement is needed here? I don't think including it has any notable affect.

Reply:
fixed